.github/workflows/android.yml

```diff
@@ -0,0 +1,79 @@
+name: Android Example App Deployment 🛫
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.ref }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  NODE_VERSION: '18'
+  JAVA_VERSION: '21'
+  JAVA_DIST: 'temurin'
+  EXAMPLE_DIRECTORY: examples/react-native-example
+  ANDROID_DIRECTORY: examples/react-native-example/android
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+      - name: Checkout 🎾
+        uses: actions/checkout@v4
+
+      - name: Setup Node 📚
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+
+      - name: Install Node modules 🎁
+        run: yarn install --frozen-lockfile
+
+      - name: Run typecheck 🎲
+        run: yarn types-check
+
+      - name: Run lint 🧹
+        run: yarn lint
+
+      - name: Run test 🏎️
+        run: yarn test
+
+      - name: Build package 🥁
+        run: yarn build
+
+      - name: Check build 🕵️‍♂️
+        run: yarn check-build
+
+      - name: Install Node modules for App 🎁
+        working-directory: ${{ env.EXAMPLE_DIRECTORY }}
+        run: yarn install --frozen-lockfile
+
+      - name: Validate Gradle ✅
+        uses: gradle/actions/wrapper-validation@v3
+
+      - name: Setup Java ♨️
+        uses: actions/setup-java@v4
+        with:
+          distribution: ${{ env.JAVA_DIST }}
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Build Android App 🧩
+        working-directory: ${{ env.ANDROID_DIRECTORY }}
+        run: ./gradlew assembleRelease
+        env:
+          BUILD_NUMBER: ${{ github.run_number }}
+
+      - name: Upload Android App to Firebase App Distribution 🏁
+        uses: wzieba/Firebase-Distribution-Github-Action@v1
+        with:
+          appId: ${{ secrets.FIREBASE_ANDROID_APP_ID }}
+          serviceCredentialsFileContent: ${{ secrets.FIREBASE_CREDENTIAL_FILE_CONTENT }}
+          groups: testers
+          releaseNotes: ${{ github.event.pull_request.title || github.ref_name }}\n${{ github.event.head_commit.message }}
+          file: ${{ env.ANDROID_DIRECTORY }}/app/build/outputs/apk/release/app-release.apk
```

Comment:
just a suggestion, for the cleanup step, we can also delete the built apk after this step.

Reply:
I don't think it's really necessary, bc android/build folder is generated one, and does not being stored on the server